pkg/providers/vsphere/contentlibrary/content_library_utils.go

```diff
@@ -184,8 +184,8 @@ func UpdateVmiWithVirtualMachine(
 			}
 
 			if uuid != "" {
-				di, _ := vmdk.GetVirtualDiskInfoByUUID(
-					ctx, nil, vm, false, uuid)
+				di, _ := virtualmachine.GetVirtualDiskInfoByUUID(
```

Comment:
VMIs can have snapshots (VMIs shouldn't even have delta disks), so I think here we need to calculate the entire chain.  Now, it doesn't matter because I only expect one disk in the chain for the VMIs, but still worth checking with @akutz to confirm that we are not missing anything.

Reply:
VMIs as OVFs will not have chains, but project-scoped libraries (backed by VMs) could, so yes, we will need the entire chain. Or at least the final node in it with a size that represents the totals of the chain.